cdap-app-fabric/src/main/java/co/cask/cdap/app/mapreduce/MRJobClient.java

```diff
@@ -61,11 +62,12 @@ public MRJobClient(CConfiguration cConf, Configuration hConf) {
 
   /**
    * @param runId for which information will be returned.
+   * @param runRecord for the run (we already keep track of information such as job-level state, start/end time).
    * @return a {@link MRJobInfo} containing information about a particular MapReduce program run.
    * @throws IOException if there is failure to communicate through the JobClient.
    * @throws NotFoundException if a Job with the given runId is not found.
    */
-  public MRJobInfo getMRJobInfo(Id.Run runId) throws IOException, NotFoundException {
+  public MRJobInfo getMRJobInfo(Id.Run runId, RunRecord runRecord) throws IOException, NotFoundException {
```

Comment:
the job client shouldn't aware of `RunRecord`. The client should be purely just for talking to mr job. Blending of RunRecord for additional / missing information should be outside of this class.